src/collection.ts

```diff
@@ -41,28 +48,105 @@ export class Collection {
     return `[Collection: ${this.path}]`;
   }
 
-  get subcollections(): Array<Collection> {
-    const collections = [];
-
-    // Find all of the sub collections.
-    const subcollectionPaths = glob.sync(`**/${Collection.ConfigFile}`, {
-      cwd: this.pod.root,
-      root: this.pod.root,
-      // Ignore the current collection.
-      // Need to strip of first / for glob matching.
-      ignore: this.collectionPath.replace(/^[/]+/, ''),
-      nodir: true,
+  /**
+   * Lists collections using glob patterns, as outlined by the [`glob`
+   * module](https://github.com/isaacs/node-glob#glob-primer).
+   *
+   * Various techniques can be used to list collections depending on your needs:
+   *
+   * ```
+   * // All top-level collections:
+   * Collection.list(pod, '/content/**')
+   *
+   * // All top-level collections, sorted by the field "order":
+   * Collection.list(pod, '/content/**', {sort: 'order'})
+   *
+   * // Both the "pages" and "posts" collections:
+   * Collection.list(pod, ['/content/posts/*', '/content/pages/*'])
+   *
+   * @param patterns A list of glob patterns or a single glob pattern. If
+   * nothing is supplied, all docs within the pod will be returned.
+   */
+  static list(
+    pod: Pod,
+    patterns?: string[] | string,
+    options?: CollectionListOptions
+  ) {
+    const paths: Set<string> = new Set();
+    if (typeof patterns === 'string') {
+      patterns = [patterns];
+    }
+    patterns = patterns || [
+      `${Pod.DefaultContentPodPath}**/${Collection.ConfigFile}`,
+    ];
+    patterns.forEach(pattern => {
+      glob
+        .sync(pattern, {
+          cwd: pod.root,
+          root: pod.root,
+          matchBase: false,
+          nodir: true,
+        })
+        .forEach(path => {
+          paths.add(path);
+        });
     });
 
-    for (const collectionPath of subcollectionPaths) {
-      collections.push(
-        new Collection(this.pod, fsPath.dirname(collectionPath))
-      );
-    }
+    // Normalize paths returned by glob. Depending on the glob pattern, the
+    // resulting paths may or may not include the pod root.
+    const cleanPaths = Array.from(paths)
+      .map(path => {
+        if (!path.startsWith(pod.root)) {
+          path = fsPath.join(pod.root, path);
+        }
+        return path.replace(pod.root, '');
+      })
+      .filter(path => {
```

Comment:
If you do the filter first then it doesn't need to map the path replacement from above on as many items.

Reply:
Good call -- thank you. :) By the way, I realized we may want to allow options to the `subcollections` getter. If I run into a scenario where this is needed I'll let you know and we can discuss changing it.